client/client.js

```diff
@@ -62,71 +65,119 @@ Client.create = function (message) {
 // Instance methods
 
 // Persist subscriptions and presences when not already persisted in memory
-Client.prototype.storeData = function (message) {
-  var subscriptions = this.subscriptions,
-      presences = this.presences,
-      changed = false,
-      lookupMessage;
+Client.prototype.storeData = function (messageIn) {
+  var subCount = Object.keys(this.subscriptions).length,
+      presCount = Object.keys(this.presences).length,
+      processedOp = false;
+
+  // Persist the message data, according to type
+  switch(messageIn.op) {
+    case 'unsubscribe':
+    case 'sync':
+    case 'subscribe':
+      processedOp = this._storeDataSubscriptions(messageIn);
+      break;
+
+    case 'set':
+      processedOp = this._storeDataPresences(messageIn);
+      break;
+  }
+
+  // FIXME: For now log everything Later, enable sample logging. 
+  // if (processedOp && subCount % LOG_WINDOW_SIZE === 0) {
+  if (processedOp) {
+    log.info('#storeData', { 
+      client_id: this.id,
```

Comment:
@jden, I don't think I understand why this is wrong. 

I decided to pass an object to the logger to get it in json format later (in the log files), so they are easier to parse. 

Are you saying that even when processedOp is false, the object is still allocated? I could probably create a logState function to avoid that allocation.